develop/blockchains/custom-blockchains/benchmarking.md

```diff
@@ -1,4 +1,191 @@
+
+There are a couple practices to notice: 
+
+- The `#[extrinsic_call]` macro is used when calling the extrinsic itself. This macro is a required part of a benchmarking function, [see the Rust docs](https://paritytech.github.io/polkadot-sdk/master/frame_benchmarking/v2/index.html#extrinsic_call-and-block){target=_blank} for more clarification
+- The `assert_eq` expression ensures that the extrinsic is working properly within the benchmark context
+
+## Adding Benchmarks to Runtime
+
+The last step before running the benchmarking tool is to ensure that the benchmarks are configured as part of your runtime.
+
+Create another file in the node's directory called `benchmarks.rs`. This is where the pallets you wish to benchmark will be registered. This file should contain the following macro, which registers all pallets for benchmarking, as well as their respective configurations: 
+
+```rust
+frame_benchmarking::define_benchmarks!(
+ [frame_system, SystemBench::<Runtime>]
+ [pallet_balances, Balances]
+ [pallet_session, SessionBench::<Runtime>]
+ [pallet_timestamp, Timestamp]
+ [pallet_message_queue, MessageQueue]
+ [pallet_sudo, Sudo]
+ [pallet_collator_selection, CollatorSelection]
+ [cumulus_pallet_parachain_system, ParachainSystem]
+ [cumulus_pallet_xcmp_queue, XcmpQueue]
+);
+```
+
+For example, if the pallet named `pallet_template` is ready to be benchmarked, it may be added as follows: 
+
+```rust
+frame_benchmarking::define_benchmarks!(
+    // ...
+ [pallet_template, Template]
+    // ...
+);
+```
+
+!!!warning 
+ If the pallet isn't included in the `define_benchmarks!` macro, the CLI will not be able to access it and benchmark it later.
+
+Navigate to the runtime's `lib.rs` file, and add the import for `benchmarks.rs` as follows: 
+
+```rust
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarks;
+```
+
+!!!info 
+ The `runtime-benchmarks` feature gate ensures that the benchmarking operations aren't a part of the production runtime. 
+
+## Running Benchmarks
+
+You can now compile your node with the following command with the `runtime-benchmarks` feature flag:
+
+```bash
+cargo build --features runtime-benchmarks --release
+```
+
+Once it is compiled with the correct feature set, you can run the benchmarking tool:
+
+```sh
+./target/release/<node-binary-name> benchmark pallet \
+--runtime <path-to-wasm-runtime> \
+--pallet <name-of-the-pallet> \
+--extrinsic '*' \
+--steps 20 \
+--repeat 10 \
+--output weights.rs
+```
+
+- `--runtime` - The path to your runtime's Wasm
+- `--pallet` - The name of the pallet you wish to benchmark. This pallet must be configured in your runtime and defined in `define_benchmarks`
+- `--extrinsic` - Which extrinsic to test. Using `*` implies all extrinics will be benchmarked
+- `--output` - Where the output of the auto-generated weights will reside
+
+The result should be a `weights.rs` file containing the types you can use to annotate your extrinsic with the correctly balanced weights in your runtime.
+
```

Comment:
After running the benchmarks you could add a section like `Adding weights to your extrinsics` and show how to import the calculated weights and assign them to each extrinsic.